From a23e90705291962dcdd08e1d00fb9a0638655d4e Mon Sep 17 00:00:00 2001
From: William Emerison Six <billsix@gmail.com>
Date: Sun, 6 Sep 2026 16:32:16 +0000
Subject: [PATCH] docs: add doc-region markers for the book's literalinclude

Comment-only markers (// doc-region-begin/end <name>) so the Mario 64 book can
literalinclude exact spans by name instead of drift-prone line numbers. First
region: euler_zxy_to_matrix (the Euler-angle rotation matrix builder).
---
 src/engine/math_util.c          | 4 ++++
 src/game/camera.h               | 2 ++
 src/game/rendering_graph_node.c | 4 ++++
 3 files changed, 10 insertions(+)

diff --git a/src/engine/math_util.c b/src/engine/math_util.c
index a5a491b9..e3b0f8e2 100644
--- a/src/engine/math_util.c
+++ b/src/engine/math_util.c
@@ -276,6 +276,7 @@ void mtxf_lookat(Mat4 mtx, Vec3f from, Vec3f to, s16 roll) {
  * Build a matrix that rotates around the z axis, then the x axis, then the y
  * axis, and then translates.
  */
+// doc-region-begin euler_zxy_to_matrix
 void mtxf_rotate_zxy_and_translate(Mat4 dest, Vec3f translate, Vec3s rotate) {
     FrameInterpolation_RecordMatrixPosRotZXY(dest, translate, rotate);
     register f32 sx = sins(rotate[0]);
@@ -305,6 +306,7 @@ void mtxf_rotate_zxy_and_translate(Mat4 dest, Vec3f translate, Vec3s rotate) {
     dest[0][3] = dest[1][3] = dest[2][3] = 0.0f;
     dest[3][3] = 1.0f;
 }
+// doc-region-end euler_zxy_to_matrix
 
 /**
  * Build a matrix that rotates around the x axis, then the y axis, then the z
@@ -498,6 +500,7 @@ void mtxf_align_terrain_triangle(Mat4 mtx, Vec3f pos, s16 yaw, f32 radius) {
  * The resulting matrix represents first applying transformation b and
  * then a.
  */
+// doc-region-begin mtxf_mul_affine
 void mtxf_mul(Mat4 dest, Mat4 a, Mat4 b) {
     // TODO: FrameInterpolation is broken, fix it
     // FrameInterpolation_RecordMatrixMult(dest, dest, 0);
@@ -543,6 +546,7 @@ void mtxf_mul(Mat4 dest, Mat4 a, Mat4 b) {
 
     mtxf_copy(dest, temp);
 }
+// doc-region-end mtxf_mul_affine
 
 /**
  * Set matrix 'dest' to 'mtx' scaled by vector s
diff --git a/src/game/camera.h b/src/game/camera.h
index 4d16db8d..e208763f 100644
--- a/src/game/camera.h
+++ b/src/game/camera.h
@@ -98,6 +98,7 @@
 #define CAM_ANGLE_MARIO  1
 #define CAM_ANGLE_LAKITU 2
 
+// doc-region-begin camera_modes
 #define CAMERA_MODE_NONE              0x00
 #define CAMERA_MODE_RADIAL            0x01
 #define CAMERA_MODE_OUTWARD_RADIAL    0x02
@@ -113,6 +114,7 @@
 #define CAMERA_MODE_8_DIRECTIONS      0x0E // AKA Parallel Camera, Bowser Courses & Rainbow Ride
 #define CAMERA_MODE_FREE_ROAM         0x10
 #define CAMERA_MODE_SPIRAL_STAIRS     0x11
+// doc-region-end camera_modes
 
 #define CAM_MOVE_RETURN_TO_MIDDLE       0x0001
 #define CAM_MOVE_ZOOMED_OUT             0x0002
diff --git a/src/game/rendering_graph_node.c b/src/game/rendering_graph_node.c
index 6f19801d..3db5581d 100644
--- a/src/game/rendering_graph_node.c
+++ b/src/game/rendering_graph_node.c
@@ -343,6 +343,7 @@ void geo_process_switch(struct GraphNodeSwitchCase *node) {
 /**
  * Process a camera node.
  */
+// doc-region-begin camera_view_matrix
 void geo_process_camera(struct GraphNodeCamera *node) {
     Mat4 cameraTransform;
     Mtx *rollMtx = alloc_display_list(sizeof(*rollMtx));
@@ -371,6 +372,7 @@ void geo_process_camera(struct GraphNodeCamera *node) {
     gMatStackIndex--;
     FrameInterpolation_RecordCloseChild();
 }
+// doc-region-end camera_view_matrix
 
 /**
  * Process a translation / rotation node. A transformation matrix based
@@ -378,6 +380,7 @@ void geo_process_camera(struct GraphNodeCamera *node) {
  * the float and fixed point matrix stacks.
  * For the rest it acts as a normal display list node.
  */
+// doc-region-begin compose_local_onto_parent
 void geo_process_translation_rotation(struct GraphNodeTranslationRotation *node) {
     Mat4 mtxf;
     Vec3f translation;
@@ -399,6 +402,7 @@ void geo_process_translation_rotation(struct GraphNodeTranslationRotation *node)
     gMatStackIndex--;
     FrameInterpolation_RecordCloseChild();
 }
+// doc-region-end compose_local_onto_parent
 
 /**
  * Process a translation node. A transformation matrix based on the node's

base-commit: 49c5312a0f3c0a28e1974be1923babd4f869f719
-- 
2.55.0

